quickwit-storage/src/local_file_storage.rs

```diff
@@ -146,6 +154,12 @@ fn missing_file_is_ok(io_result: io::Result<()>) -> io::Result<()> {
     }
 }
 
+impl From<PathBuf> for LocalFileStorage {
```

Comment:
I think sometime back we wanted to only have the LocalFileStorage created with a check on whether the path is rooted or not `Self::extract_root_path_from_uri(uri)?;` to protect against. we could possibly have that check here or make add a comment for people using `LocalFileStorage::from` in another context.